Add _latex methods for DifferentialOperator , GroebnerBasis

Lenqth opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234". See
https://github.com/blog/1506-closing-issues-via-pull-requests .-->


#### Brief description of what is fixed or changed

Add _latex methods for GroebnerBasis and DifferentialOperator
because they are printed like "t**3-x,t*x-y,t*y-?z,t*z-x**2,x**3..." in Jupyter

#### Other comments


#### Release Notes

<!-- Write the release notes for this release below. See
https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more information
on how to write release notes. The bot will check your release notes
automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->

* holonomic
   * Add a _latex method for DifferentialOperator class

* polys
   * Add a _latex method for GroebnerBasis class

* interactive
   * Add empty class Printable just for mark class as printable_types
   * Impremented TODO about hasattr(o, '_latex') in _can_print_latex

<!-- END RELEASE NOTES -->

Lenqth committed:
Add _latex printing method for DifferentialOperator , GroebnerBasis

jksuom commented on sympy/holonomic/holonomic.py:
```diff
@@ -135,7 +136,7 @@ def __eq__(self, other):
 
 
-class DifferentialOperator(object):
+class DifferentialOperator(Basic):
```
Is it necessary to make this derive from `Basic`?

jksuom replied:
It *could* be possible to derive this from Basic (with some trouble) but that looks like a hack to me. A better way might be implementing this [TODO](https://github.com/sympy/sympy/blob/master/sympy/interactive/printing.py#L126).

Lenqth replied:
Thank you for the advice.
I impremented the TODO, and now DifferentialOperator derive from new empty class Printable ( just for printable_types ) instead of Basic.

Lenqth committed:
Fix _latex's argument name.

Lenqth committed:
[fix] Fix trouble caused by DifferentialOperator had been derived from Basic

Added empty class Printable for mark class as printable_types
Impremented TODO about hasattr(o, '_latex') in _can_print_latex

DifferentialOperator is now derived from Printable

asmeurer commented:
Sorry for the duplicate comments. GitHub was not being responsive yesterday due to its [outage issues](https://blog.github.com/2018-10-21-october21-incident-report/), and it appeared that my comments did not go through, but apparently they did.

asmeurer reviewed (changes requested):
This needs tests as well.

asmeurer commented on sympy/holonomic/holonomic.py:
```diff
@@ -371,6 +373,32 @@ def is_singular(self, x0):
+            if v == self.parent.base.zero:
+                continue
+
+            lv = latex(v)
```
Use `printer`.

Lenqth committed:
_latex uses printer._print() instead of latex()
remove importing of latex
fix latex printing format of GroebnerBasis (\mathrm)

czgdp1807 commented on sympy/polys/polytools.py:
```diff
@@ -7036,6 +7036,13 @@ def contains(self, poly):
+        gens = ",".join( [ printer._print(gen) for gen in self.gens] )
+        dm = printer._print(self.domain)
+        order = printer._print(self.order)
+        return r"\mathrm{GroebnerBasis}([%s],%s,\mathrm{domain=}%s,\mathrm{order=}%s)" % (pls,gens,dm,order)
```
Please break this line into two. If possible.

czgdp1807 commented:
@Lenqth Are you still working on it? Please resolve the conflicts and make the requested changes.
